binding/react/src/use_rhino.ts

```
@@ -9,277 +9,132 @@
+      }
 
-    if (rhinoWorker !== null) {
-      rhinoWorker.onmessage = (
-        msg: MessageEvent<RhinoWorkerResponse>
-      ): void => {
-        switch (msg.data.command) {
-          case 'rhn-inference':
-            setIsTalking(false);
-            rhinoWorker.postMessage({ command: 'pause' });
-            // We know this inference isFinalized, so assert to more specific type
-            inferenceCallback(msg.data.inference as RhinoInference);
-            break;
-          case 'rhn-error':
-            setIsError(true);
-            setErrorMessage(msg.data.error.toString());
-            break;
-          case 'rhn-info':
-            setContextInfo(msg.data.info);
-            break;
-          default:
-            break;
+      try {
+        if (!rhinoRef.current) {
+          rhinoRef.current = await RhinoWorker.create(
+            accessKey,
+            context,
+            inferenceCallback,
+            model,
+            { ...options, processErrorCallback: errorCallback }
+          );
+          setContextInfo(rhinoRef.current.contextInfo);
+          setIsLoaded(true);
+          setError(null);
         }
-      };
-    }
-  }, [inferenceCallback]);
-
-  /** Startup (and cleanup) Rhino */
-  useEffect(() => {
-    // Reset hook state, since we're rebooting Rhino and WebVoiceProcessor
-    setIsTalking(rhinoHookArgs?.isTalking === true);
-    setIsListening(false);
-    setIsLoaded(false);
-    setIsError(false);
-    setErrorMessage(null);
-
-    if (rhinoWorkerFactory === null || rhinoWorkerFactory === undefined) {
-      return (): void => {
-        /* NOOP */
-      };
-    }
-
-    if (rhinoHookArgs === null || rhinoHookArgs === undefined) {
-      return (): void => {
-        /* NOOP */
-      };
+      } catch (e: any) {
+        errorCallback(e.toString());
+      }
+    },
+    [inferenceCallback]
+  );
+
+  const process = useCallback(async (): Promise<void> => {
+    try {
+      if (rhinoRef.current && !isListening) {
```

Might be good to check for state problems (fail loud):
```typescript
if (!rhinoRef.current) {
  setError('Rhino has not been initialized or has been released');
  return;
}
```